src/Workspaces/Core/Portable/Shared/Extensions/ITypeSymbolExtensions.cs

```diff
@@ -167,9 +168,10 @@ public static ITypeSymbol RemoveNullableIfPresent(this ITypeSymbol symbol)
             // if they are considered to be the same type, which provides a more accurate
             // implementations list for interfaces. 
             var typeSymbolProject = solution.GetProject(typeSymbolAndProjectId.ProjectId);
-            var typeSymbolCompilation = typeSymbolProject == null ?
-                                        null :
-                                        await typeSymbolProject.GetCompilationAsync(cancellationToken).ConfigureAwait(false);
+            var interfaceMemberProject = solution.GetProject(interfaceMemberAndProjectId.ProjectId);
+
+            var typeSymbolCompilation = await GetCompilationOrNull(typeSymbolProject, cancellationToken).ConfigureAwait(false);
```

Comment:
these 2 compilations, can't be null. if null is assigned to these compilation, then we will again throw exception.

if one of compilations return null, we should either bail out or make sure OriginalSymbolsMatch can handle null compilations.

Reply:
Updated VerifyForwardedTypes to try and handle null in the same way for both compilations.